asu/util.py

```diff
@@ -368,6 +368,21 @@ def parse_feeds_conf(url: str) -> list[str]:
     )
 
 
+def is_new_style_build(version: str) -> bool:
+    """For imagebuilder containers using 'setup.sh' instead of fully populated."""
+    if version.lower().endswith("snapshot"):
+        return True
+
+    if version.startswith("24."):
```

Comment:
The "new style" only applies to snapshots aka daily changing containers. Tagged versions don't change and contain the full image builder.

Reply:
The rc builds are using it, too:

```
$ podman run -it ghcr.io/openwrt/imagebuilder:x86-64-v24.10.0-rc1
buildbot@d2ddfc54230f:~$ ls -la
total 28
drwxr-xr-x 3 buildbot buildbot 4096 Dec  2 10:35 .
dr-xr-xr-x 1 root     root     4096 Dec  3 17:01 ..
-rw-r--r-- 1 buildbot buildbot  220 Mar 27  2022 .bash_logout
-rw-r--r-- 1 buildbot buildbot 3526 Mar 27  2022 .bashrc
drwx------ 3 buildbot buildbot 4096 Dec  2 10:35 .gnupg
-rw-r--r-- 1 buildbot buildbot  807 Mar 27  2022 .profile
-rwxr-xr-x 1 root     root      925 Dec  2 10:35 setup.sh
```

Reply:
There are actually two things changed, the kmod-split and the setup.sh-based containers.  Currently they overlap completely, with both applying to

1) All main SNAPSHOT and all release -SNAPSHOT builds.
2) All 24.10* and later.
3) As-yet unreleased 23.05.6 and later on that branch (I suppose any 22.x built after today, too).

What's the plan?  I suspect keep kmod-split as-is, but remove the setup.sh-based containers from all but SNAPSHOTs?

Reply:
Is this the culprit for the setup.sh in the RC images?  

https://github.com/openwrt/docker/blob/main/Dockerfile#L26

Does it need some sort of logic like `ARG RUN_SETUP=!version.endswith("snapshot")` however you'd say that inside a docker file???

Reply:
Should be fixed now, thanks for letting me know

Reply:
Will the already-existing rc2 images be replaced?  They got generated before you got the fix in, so I'm still seeing them with the bug.

In any case, I'll rework this PR to make everything correspond with the fix and push a new version.

Reply:
Oops, unclear above: "rc2 images" should be "rc2 docker images on ghcr, quay etc."  The stuff on downloads is fine.